File: fabric-audit-agent/core/data-agent.js
/**
 * Build the Fabric Data Agent / MCP manifest from the agent's tool definitions.
 * This is the contract a Fabric Data Agent or MCP server exposes so the auditor is
 * callable from Copilot in Power BI / M365 Copilot / Copilot Studio. Pure: tools in, manifest out.
 * @param {object[]} toolDefinitions  from tools.js `createToolDefinitions()`
 */
export function buildDataAgentManifest(toolDefinitions = []) {
  return {
    name: 'fabric-audit-agent',
    displayName: '[C] Fabric Audit Agent',
    description: 'Read-only Microsoft Fabric / Power BI capacity & performance advisor. Ask it to audit the estate or explain an issue.',
    instructions: 'Call run_audit to sweep the estate and return prioritized findings, a digest, and the capacity verdict. The agent is strictly read-only.',
    readOnly: true,
    tools: toolDefinitions.map(t => ({
      name: t.name,
      description: t.description,
      input_schema: t.input_schema,
    })),
  };
}

File: fabric-audit-agent-py/fabric_audit_agent/data_agent.py
"""Fabric Data Agent / MCP manifest from tool definitions. Port of ``core/data-agent.js``. Pure."""


def build_data_agent_manifest(tool_definitions=None):
    tool_definitions = tool_definitions or []
    return {
        "name": "fabric-audit-agent",
        "displayName": "[C] Fabric Audit Agent",
        "description": "Read-only Microsoft Fabric / Power BI capacity & performance advisor. Ask it to audit the estate or explain an issue.",
        "instructions": "Call run_audit to sweep the estate and return prioritized findings, a digest, and the capacity verdict. The agent is strictly read-only.",
        "readOnly": True,
        "tools": [{"name": t.get("name"), "description": t.get("description"), "input_schema": t.get("input_schema")} for t in tool_definitions],
    }
